File: Greedy/NonOverlapIntervals/over.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minRemoval(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());

        int currEnd = intervals[0][1];
        int n = intervals.size();
        int removals = 0;

        for(int i = 1; i < n; i++) {
            int nextStart = intervals[i][0];
            int nextEnd = intervals[i][1];

            if(nextStart < currEnd) {
                removals++;
                currEnd = min(currEnd, nextEnd);
            }
            else {
                currEnd = nextEnd;
            }
        }

        return removals;
    }
};

File: Greedy/NonOverlapIntervals/over.py
class Solution:
    def minRemoval(self, intervals):
        # code here
        intervals.sort()
        currEnd = intervals[0][1]
        n = len(intervals)
        removals = 0
        for i in range(1,n):
            nextStart = intervals[i][0]
            nextEnd = intervals[i][1]
            if nextStart < currEnd:
                removals += 1
                currEnd = min(currEnd,nextEnd)
            else:
                currEnd = nextEnd
        return removals